Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -658,47 +632,43 @@ this has no time dependence prior to this operation.
  * ignored, i.e., no time index is added.
 **/
 void ComponentInfo::merge(const ComponentInfo &other) {
-  checkNoTimeDependence();
-  const auto &toMerge = buildMergeIndicesSync(other);
-  for (size_t timeIndex = 0; timeIndex < other.m_scanIntervals->size();
+  const auto &toMerge = buildMergeIndices(other);
+  // Merging the detectorInfo has to be done before we update scanIntervals
+  m_detectorInfo->merge(*other.m_detectorInfo, toMerge);
+  for (size_t timeIndex = 0; timeIndex < other.m_scanIntervals.size();
        ++timeIndex) {
     if (!toMerge[timeIndex])
       continue;
-    auto &scanIntervals = m_scanIntervals.access();
     auto &positions = m_positions.access();
     auto &rotations = m_rotations.access();
-    scanIntervals.push_back((*other.m_scanIntervals)[timeIndex]);
+    m_scanIntervals.push_back(other.m_scanIntervals[timeIndex]);
```

Comment:
It's probably not worth doing anything about, but there are operations here, such as this one,that could fail after the detector infos have been merged.